### ISSUE TITLE
Where do I get the information needed to fill in Config?

### ISSUE THREAD

**Issue by entreprenewer** [lprhodes/homebridge-broadlink-rm#206]
Hi
So sorry for an extremely basic question, but I've poured over the ReadMe and sample config files, and for the life of me, I can't figure out how and where to get the values that I need to input in the Config for my devices. The Readme says my devices should automatically be added, and that when I press a button on a remote and point to the device, it should show up in Terminal console - but it doesn't. I've tried flipping "Learn" on on my iPhone, and it scans for 5s, I press on the remote, but nothing.

Thanks for any guidance anyone can offer.

**Comment by RickHOZ** [comment]
Hi, I'm just about to go through the same stage, just awaiting delivery of my Broadlink.

I was also under the assumption from reading the materials that it was supposed to just collect the device info. I will be watching this thread hoping somebody helps you. It seems very quiet in here I have had an issue listed for a few days and have not heard anything back.

Hopefully, some of the guys are still around and willing to help.

**Comment by RickHOZ** [comment]
Hi All,

Broadlink received so I would appreciate any help that might be available.

entreprenewer ? Did you get a resolution?

**Comment by entreprenewer** [comment]
Unfortunately not. Nobody has replied.

**Comment by RickHOZ** [comment]
Hi entreprenewer,  Well after a major amount of jigging around I managed to get mine to work and I now have it learning IR.

If you want to copy and paste your config.json file I will have a look and see if I can make the same changes.

I am so bloody happy after days of playing around with this.

the other way would be open a git an send me the link.

RickHOZ

**Comment by kyc1338** [comment]
My Homebridge detected my 4 RMs but can't learn IR or RF. Can anyone help?

**Comment by RickHOZ** [comment]
Hi KYC1338

I am by no means an expert. I have got mine to run, through a lot of trial and error. If you want to paste your code in or open a new git I will have a look. No promises, however.

**Comment by kyc1338** [comment]
Hi RickHOZ

Thanks for your prompt response. My config.json is the same as the sample for multi RMs and only changed the IP for RM.

I got this in my log file

[11/25/2017, 9:10:49 AM] [Broadlink RM] Scan RF (RF learning not supported for device at undefined)
[11/25/2017, 9:09:43 AM] [Broadlink RM] Scan RF (RF learning not supported for device at undefined)
[11/25/2017, 9:09:39 AM] [Broadlink RM] Living Room Switch getSwitchState: 0
[11/25/2017, 9:09:39 AM] [Broadlink RM] Dining room Switch getSwitchState: 0
[11/25/2017, 9:07:07 AM] [Broadlink RM] Learn Code (stopped)
[11/25/2017, 9:07:07 AM] [Broadlink RM] Learn Code (stopped - 10s timeout)
[11/25/2017, 9:06:57 AM] [Broadlink RM] Learn Code (ready)
[11/25/2017, 9:06:52 AM] [Broadlink RM] Living Room Switch getSwitchState: 0
[11/25/2017, 9:06:52 AM] [Broadlink RM] Dining room Switch getSwitchState: 0
[11/24/2017, 10:58:52 PM] [Broadlink RM] Living Room Switch getSwitchState: 0
[11/24/2017, 10:58:52 PM] [Broadlink RM] Dining room Switch getSwitchState: 0
[11/24/2017, 10:40:22 PM] [Broadlink RM] Learn Code (stopped)
[11/24/2017, 10:40:22 PM] [Broadlink RM] Learn Code (stopped - 10s timeout)
[11/24/2017, 10:40:11 PM] [Broadlink RM] Learn Code (ready)
[11/24/2017, 10:39:39 PM] [Broadlink RM] Learn Code (stopped)
[11/24/2017, 10:39:32 PM] [Broadlink RM] Learn Code (ready)
[11/24/2017, 10:37:24 PM] [Broadlink RM] Scan RF (RF learning not supported for device at undefined)
[11/24/2017, 10:37:20 PM] [Broadlink RM] Living Room Switch getSwitchState: 0
[11/24/2017, 10:37:20 PM] [Broadlink RM] Dining room Switch getSwitchState: 0
[11/24/2017, 10:29:58 PM] [Broadlink RM] Learn Code (stopped)
[11/24/2017, 10:29:58 PM] [Broadlink RM] Learn Code (stopped - 10s timeout)
[11/24/2017, 10:29:48 PM] [Broadlink RM] Learn Code (ready)
[11/24/2017, 10:29:43 PM] [Broadlink RM] Living Room Switch getSwitchState: 0
[11/24/2017, 10:29:43 PM] [Broadlink RM] Dining room Switch getSwitchState: 0
[11/24/2017, 10:17:33 PM] [Broadlink RM] Living Room Switch getSwitchState: 0
[11/24/2017, 10:17:33 PM] [Broadlink RM] Dining room Switch getSwitchState: 0
Discovered Broadlink RM device at 192.168.100.152 (0d:43:b4:aa:4a:e8)
Discovered Broadlink RM device at 192.168.100.191 (0d:43:b4:30:53:a3)
Discovered Broadlink RM device at 192.168.100.153 (0d:43:b4:38:fc:5d)
Discovered Broadlink RM device at 192.168.100.103 (0d:43:b4:cc:38:f7)
[11/23/2017, 10:23:03 PM] Homebridge is running on port 51826.

**Comment by RickHOZ** [comment]
Hi,

Sorry i'm out the door on a trip and have not got time to rerspond until next week. Anyway have a look at this copy and paste this is a working file on my system Maybe just delete out what you don't want and use this. I ran the code from the sample (config-multiple-rm-devices-sample.json)  in json formatter and it did not work. (https://jsonformatter.curiousconcept.com)

So dlete out what you don't want then run it through Json formatter to check it then try again.

good luck.

{
   "bridge":{
      "name":"Samara",
      "username":"CC:22:3D:E3:CE:30",
      "port":51826,
      "pin":"031-45-154"
   },
   "description":"My home HomeKit API configuration file",
   "platforms":[
      {
         "platform":"ElkM1",
         "name":"ElkM1",
         "elkAddress":"192.168.1.2",
         "elkPort":"2101",
         "area":"1",
         "keypadCode":"XXXX",
         "zoneTypes":{
           "4":"motion",
           "5":"motion",
           "7":"motion",
           "8":"motion",
           "9":"motion",
           "10":"motion",
           "11":"motion",
           "12":"motion",
           "14":"motion",
           "15":"motion",
           "16":"motion",
           "17":"motion",
           "18":"motion",
           "19":"motion",
           "20":"motion",
           "21":"motion",
           "22":"motion",
           "23":"motion",
           "24":"motion",
           "25":"motion",
           "26":"motion",
           "28":"motion",
           "29":"motion"
         },
         "garageDoors":[
            {
               "stateZone":"5",
               "openOutput":"11",
               "closeOutput":"11",
               "name":"Garage door"
            }
         ]
      },
      {
        "platform":"BroadlinkRM",
        "name":"Broadlink RM",
        "hideScanFrequencyButton": false,
        "hideLearnButton": false,
        "accessories":[  {
          "name":"Apple TV Menu",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"2600580000012a94141313381437143714121437143714371437153714371412141214131412143714371437141314121412141313131412143714371413141214371412141214131400051900012b4914000c6300012a4914000d05"
      },
      {
          "name":"Apple TV Play",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"2600500000012a94141214371437143714131437143714371437143714371413141214121412143814371412143714131412141214121413143714371412141214371512141214121400051e00012b4914000d050000000000000000"
      },
      {
          "name":"Apple TV Up",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"2600500000012a94141214371438143714121437143714371437143714371512141214121413143714121437141314371412141214131313143714371412141313381412141214131300051a00012a4914000d050000000000000000"
     },
     {
          "name":"Apple TV Down",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"2600500000012995141314371437143714121437143714381338143714371412141313131412143714131313143714371412141314121412143714371413141214371412141314121400051e00012a4914000d050000000000000000"
      },
      {
          "name":"Apple TV Left",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"2600580000012a94141214371437143714131338133813381437143714371412141313131412143813381313131314371412141313131412143714371413131313381313141214131300051a0001294a13000c620001294a13000d05"
      },
      {
          "name":"Apple TV Right",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"260058000001289612141239123a113a11151239123912391239123912391215111512141214123a12141239123912151115121412141215113a113a121412141239121511151214120005200001284b12000c650001274c12000d05"
      },
      {
          "name":"Marantz On",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"26006c001e1d3b393b1c1e1d1d1d1e901e1d1e391e1c3b1c1e1d1e1c1e1c1e1c1e1d1e391e0009ba1e1d3b393b1d1d1d1e1c1e911e1c1e391e1c3b1d1e1c1e1c1e1c1e1d1e1c1e391e0009bb1e1c3b393b1d1e1c1e1c1e911e1c1e391e1d3b1c1e1c1e1c1e1d1e1c1e1c1e391e000d05000000000000000000000000"
      },
      {
          "name":"Marantz Off",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"26006c001e1c1e1d1e1c3b1c1e1c1e1d1e901e1c1f391e1c3b1c1e1d1e1c1e1c1e1c1e393c0009d71f1c1e1c1e1c3b1d1e1c1e1c1e911e1c1e391e1c3c1c1e1c1e1c1e1c1e1d1e393b0009d81e1c1e1c1e1d3b1c1e1c1e1c1e911e1c1e3a1e1c3b1c1e1c1e1d1e1c1e1c1e393b000d05000000000000000000000000"
      },
      {
         "name":"Marantz Apple TV",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"26004e001e1c3b1c1e1d1d3a1e1c3b391e1d1e1c1e1c1e1c1e1d1e000b6f1e1c3b1d1d1d1e391d1d3a3b1c1e1c1e1c1e1c1f1b1f1b000b721e1c3b1d1d1d1e391e1c3b3a1d1d1e1c1e1c1e1d1d1d1e000d0500000000000000000000"
      },
      {
          "name":"Marantz Foxtel",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"26004e001d1d3b1c1e1c1e3a3b391e1c1e1d1d1d1e1c1e1c1e1d1d000b701e1c3b1c1e1d1e393b391e1d1e1c1e1c1e1c1e1d1d1d1e000b6f1e1c3b1d1e1c1e393b3a1e1c1e1c1e1c1e1c1e1d1e1c1e000d0500000000000000000000"
      },
      {
          "name":"Marantz Volume UP",
          "type":"switch-repeat",
          "sendCount":5,
          "data":"260034001e1c1f1c1e1c3b1c1e1c1f1c1e1c1e393b1d1e1c1e1c1e000b8c1e1c1f1c1e1c3b1c1e1c1f1c1e1c1e393b1d1e1c1e1c1e000d0500000000"
      },
      {
          "name":"Marantz Volume Down",
          "type":"switch-repeat",
          "sendCount":5,
          "data":"260030001e1c3b393b1d1e1c1e1c1e1c1e3a3b1c1e1c1e391e000b701e1c3b393c1c1e1c1e1c1e1c1e3a3b1c1e1c1e3a1e000d050000000000000000"
      },
      {
          "name":"Epson On",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"2600d800000124931138123811141114111411141113113911381214113811141138121311391114111411131213111411381214111311391138123811381238121311391138121311000528000123931238113911141114111311141114113812381114113812131238111411381213121311141114111411381114111411381238113911381238111411381238111411000528000123931139113812131213111411141114113811391113123811141138121411381114111411141113121311391114111411381139113812381138121411381139111312000d05"
      },
      {
          "name":"Epson Off",
          "type":"switch-repeat",
          "sendCount":1,
          "data":"26002001000123931139113812141113111411141114113811391114113812131139111411381114113812141113111411391114111411381114113812381138121411381139111411000527000124931138123811141114111411141113113911381214113811141138121311391114113812131114111411381213111411381214113811381238121311391138121311000528000123931238113911141114111311141114113812381114113812141138111411381214123711141114111411381114111412371312113911381238111411381238111411000528000123931238123712131213111411141114123711391114113811141139121311381114113812141113111411381214111411381114123712381138121411381139111411000d050000000000000000"
          }
      ]
    },
{
         "platform":"homebridge-cbus.CBus",
         "name":"CBus",
         "client_ip_address":"127.0.0.1",
         "client_controlport":20023,
         "client_cbusname":"SAMARA",
         "client_network":254,
         "client_application":56,
         "client_debug":true,
         "platform_export":"/Users/SamaraServer/.homebridge/samara_platform.json",
         "accessories":[
            {"type":"light","id":11,"name":"Office lights"},
            {"type":"light","id":12,"name":"Office Store"},
            {"type":"light","id":74,"name":"AirCon Office"},
            {"type":"light","id":13,"name":"Office Toilet"},
            {"type":"dimmer","id":18,"name":"Office Hall"},
            {"type":"light","id":50,"name":"Office Toilet Exhaust Fan"},
            {"type":"dimmer","id":1,"name":"Master Bedroom"},
            {"type":"light","id":75,"name":"Master Bedroom AirCon"},
            {"type":"light","id":2,"name":"Master Bedroom Feature"},
            {"type":"dimmer","id":3,"name":"Master Bedroom Ensuite"},
            {"type":"light","id":4,"name":"Master Bedroom Ensuite Mirror"},
            {"type":"dimmer","id":5,"name":"Master Bedroom Toilet"},
            {"type":"light","id":6,"name":"Master Bedroom Wardrobe"},
            {"type":"light","id":48,"name":"Master Bathroom Fan"},
            {"type":"light","id":49,"name":"Master Toilet fan"},
            {"type":"dimmer","id":7,"name":"Retreat"},
            {"type":"dimmer","id":8,"name":"Retreat Cove"},
            {"type":"dimmer","id":9,"name":"Retreat Stairs"},
            {"type":"light","id":76,"name":"AirCon Retreat"},
            {"type":"light","id":145,"name":"Retreat Feature"},
            {"type":"dimmer","id":15,"name":"Media Room"},
            {"type":"dimmer","id":16,"name":"Media Cove"},
            {"type":"light","id":73,"name":"AirCon Media"},
            {"type":"dimmer","id":21,"name":"Kitchen"},
            {"type":"light","id":22,"name":"Kitchen Benches"},
            {"type":"light","id":71,"name":"AirCon Kitchen"},
            {"type":"light","id":86,"name":"Kitchen Bulkhead"},
            {"type":"light","id":23,"name":"Scullery"},
            {"type":"dimmer","id":24,"name":"Dining Room"},
            {"type":"light","id":10,"name":"House Features"},
            {"type":"dimmer","id":26,"name":"Lounge"},
            {"type":"light","id":72,"name":"AirCon Lounge"},
            {"type":"light","id":85,"name":"Fireplace"},
            {"type":"dimmer","id":40,"name":"Sala Treatment Room"},
            {"type":"dimmer","id":38,"name":"Sala"},
            {"type":"light","id":39,"name":"Sala Feature"  },
            {"type":"light","id":56,"name":"Sala Bathroom Exhaust Fan"},
            {"type":"light","id":35,"name":"Laundry"},
            {"type":"light","id":36,"name":"Laundry Store"},
            {"type":"light","id":37,"name":"Laundry Clothesline"},
            {"type":"dimmer","id":19,"name":"Laundry Hall"},
            {"type":"light","id":110,"name":"Externals Rear and Clothesline"},
            {"type":"light","id":45,"name":"Pool Pump Room"},
            {"type":"light","id":46,"name":"Wine Cellar"},
            {"type":"light","id":67,"name":"Fountain"},
            {"type":"light","id":68,"name":"Garden Lighting"},
            {"type":"light","id":69,"name":"Green Garden Lights"},
            {"type":"light","id":58,"name":"Pond Light"},
            {"type":"light","id":57,"name":"Pond Pump"},
            {"type":"light","id":70,"name":"Pool Lights"},
            {"type":"light","id":43,"name":"Deck Pool"},
            {"type":"light","id":60,"name":"Pool Pump"},
            {"type":"light","id":61,"name":"Pool Spa Pump"},
            {"type":"dimmer","id":28,"name":"Bridge Suite"},
            {"type":"dimmer","id":29,"name":"Bridge Suite Ensuite"},
            {"type":"light","id":30,"name":"Bridge Suite Ensuite Mirror"},
            {"type":"light","id":77,"name":"Bridge Suite AirCon"},
            {"type":"light","id":44,"name":"Bridge Suite Deck"},
            {"type":"light","id":53,"name":"Bridge Suite Ensuite Exhaust Fan"},
            {"type":"light","id":66,"name":"Bridge Suite AV Features"},
            {"type":"dimmer","id":31,"name":"Kingfisher Bed"},
            {"type":"light","id":78,"name":"Kingfisher Bed AirCon"},
            {"type":"dimmer","id":32,"name":"Kingfisher lounge"},
            {"type":"light","id":79,"name":"Kingfisher Lounge AirCon"},
            {"type":"dimmer","id":33,"name":"Bathroom Kingfisher"},
            {"type":"light","id":54,"name":"Bathroom Exhaust Fan Kingfisher"},
            {"type":"dimmer","id":34,"name":"Kingfisher Toilet"},
            {"type":"light","id":55,"name":"Kingfisher Toilet Exhaust Fan"},
            {"type":"dimmer","id":20,"name":"Guest Hall"},
            {"type":"light","id":47,"name":"Portico"},
            {"type":"switch","id":80,"name":"Front Door Open","activeDuration":"5s"},
            {"type":"light","id":87,"name":"Chandelier"},
            {"type":"light","id":25,"name":"Foyer"},
            {"type":"switch","id":105,"name":"Front Gate Open","activeDuration":"100ms"},
            {"type":"switch","id":103,"name":"Garage Door Open","activeDuration":"100ms"},
            {"type":"light","id":14,"name":"Garage"},
            {"type":"light","id":52,"name":"Solar Hot Water Boost"},
            {"type":"shutter","id":64, "name": "Master Bedroom Curtains", "invert": "true"},
            {"type":"shutter","id":63, "name": "Master Bedroom Net", "invert": "true"}
         ]
      }
   ],
   "accessories":[

   ]
}

**Comment by kyc1338** [comment]
My problem is I can get the code in the log file. Finally I used RM-bridge to get the code and input into the config file. Now my Home App is working.

**Comment by entreprenewer** [comment]
I'd be interested in working with someone like @RickHOZ to create some kind of manual or instructional video to contribute here to other users. I create online courses and learning materials for a living, so if someone could help me understand what needs to be done to set this up, I would happily give back and create a clear walk-through of how it's done.

I received my Raspberry Pi last week, and have two broad link devices, so it would be a pretty thorough video.

Let me know!

**Comment by RickHOZ** [comment]
Hi All, Not sure I am the best person for this as everything I have done has been more trial an error than knowledge. A lot of reading and errors. I can certainly help with reviewing anything you do and make some suggestions and adding what I have learned along the way. So if you map it out in a text form I can try and help.

It has been a challenge and a lot of fun as I solved each issue and I am amazed at what I can now do in my home. Controlling all devices using the spoken word. My wife thinks I am completely nuts. I can't wait to try the new Apple Homepod. Rick
